mypyc/test-data/run-strings.test

```diff
@@ -178,3 +148,164 @@ def test_str_to_bool() -> None:
+    assert str(Point(4, 2)) == "Point(4, 2)"
+
+    # Accessing arguments’ items:
+    coord2 = (3, 5)
+    assert 'X: {0[0]};  Y: {0[1]}'.format(coord2) == 'X: 3;  Y: 5'
+
+    # Replacing %s and %r:
+    s = "repr() shows quotes: {!r}; str() doesn't: {!s}".format('test1', 'test2')
+    assert s == "repr() shows quotes: 'test1'; str() doesn't: test2"
+
+    # Aligning the text and specifying a width:
+    assert '{:<30}'.format('left aligned') == 'left aligned                  '
+    assert '{:>30}'.format('right aligned') == '                 right aligned'
+    assert '{:^30}'.format('centered') == '           centered           '
+    assert '{:*^30}'.format('centered') == '***********centered***********' # use '*' as a fill char
+
+    # Replacing %+f, %-f, and % f and specifying a sign:
+    assert '{:+f}; {:+f}'.format(3.14, -3.14) == '+3.140000; -3.140000' # show it always
+    assert '{: f}; {: f}'.format(3.14, -3.14) == ' 3.140000; -3.140000' # show a space for positive numbers
+    assert '{:-f}; {:-f}'.format(3.14, -3.14) == '3.140000; -3.140000' # show only the minus -- same as '{:f}; {:f}'
+
+    # Replacing %x and %o and converting the value to different bases:
+    s = "int: {0:d};  hex: {0:x};  oct: {0:o};  bin: {0:b}".format(42) # format also supports binary numbers
+    assert s == 'int: 42;  hex: 2a;  oct: 52;  bin: 101010'
+    s = "int: {0:d};  hex: {0:#x};  oct: {0:#o};  bin: {0:#b}".format(42) # with 0x, 0o, or 0b as prefix:
+    assert s == 'int: 42;  hex: 0x2a;  oct: 0o52;  bin: 0b101010'
+
+    # Using the comma as a thousands separator:
+    assert '{:,}'.format(1234567890) == '1,234,567,890'
+
+    # Expressing a percentage:
+    points = 19.0
+    total = 22.0
+    assert 'Correct answers: {:.2%}'.format(points/total) == 'Correct answers: 86.36%'
+
+    # Using type-specific formatting:
+    import datetime
+    d = datetime.datetime(2010, 7, 4, 12, 15, 58)
+    assert '{:%Y-%m-%d %H:%M:%S}'.format(d) == '2010-07-04 12:15:58'
+
+    # Nesting arguments and more complex examples:
+    tmp_strs = []
+    for align, text in zip('<^>', ['left', 'center', 'right']):
+        tmp_strs.append('{0:{fill}{align}16}'.format(text, fill=align, align=align))
+    assert tmp_strs == ['left<<<<<<<<<<<<', '^^^^^center^^^^^', '>>>>>>>>>>>right']
+
+    octets = [192, 168, 0, 1]
+    assert '{:02X}{:02X}{:02X}{:02X}'.format(*octets) == 'C0A80001'
+
+    width = 5
+    tmp_strs = []
+    for num in range(5,12):
+        tmp_str = ""
+        for base in 'dXob':
+            tmp_str += ('{0:{width}{base}}'.format(num, base=base, width=width))
+        tmp_strs.append(tmp_str)
+    assert tmp_strs == ['    5    5    5  101',\
+                        '    6    6    6  110',\
+                        '    7    7    7  111',\
+                        '    8    8   10 1000',\
+                        '    9    9   11 1001',\
+                        '   10    A   12 1010',\
+                        '   11    B   13 1011']
```

Comment:
Test `{}` with bool, bytes and float arguments. Test both positive and negative floats + zero. Test a very large float and a float that is close to zero (e.g. 1.23e-20). Test infinities and NaN.